src/audit/cache_poisoning.rs

```diff
@@ -0,0 +1,292 @@
+use crate::audit::{audit_meta, WorkflowAudit};
+use crate::finding::{Confidence, Finding, Severity};
+use crate::models::Step;
+use crate::state::AuditState;
+use github_actions_models::common::expr::ExplicitExpr;
+use github_actions_models::common::{Env, EnvValue};
+use github_actions_models::workflow::event::{BareEvent, OptionalBody};
+use github_actions_models::workflow::job::StepBody;
+use github_actions_models::workflow::Trigger;
+use std::ops::Deref;
+use std::sync::LazyLock;
+
+/// The possible toggle types for cache-aware Actions.
+/// A boolean toggle means only one possible literal value (true)
+/// to opt into the cache behavior, whereas a multi-value toggle means
+/// two or more values defining opt in for caching
+#[derive(PartialEq)]
+enum CacheControlToggle {
+    Boolean,
+    MultiValue,
+}
+
+/// The general schema for a cache-aware actions
+struct CacheAwareAction {
+    /// The owner/repo part within the Action full coordinate
+    reference: &'static str,
```

Comment:
Two additional considerations here:

* We should make sure to support `owner/repo/action/path`, i.e. subpaths 🙂 
* Action defaults will probably change over time, so in practice we probably also need the ability to indicate versions here

Put together, and I _think_ it might make sense to re-use the `RepositoryUses` model here, since that'll give us all the (optional) additional state we need. I've also added `RepositoryUses::matches` in #295, which should make fuzzy-matching of different `uses:` variants slightly more convenient.

Reply:
Thanks for sharing this input!

The way I solved subpaths was by adding two different `CacheAwareAction` instances in the list, particularly in the case of `actions/cache` and `actions/cache/restore`. 

This was a simple take ofc, and the only advantage I see for sticking with it is the possibility of two Actions in two different subpaths actually defining different sets of inputs/outputs, meaning that we would need to handle them separatedly anyway.

That's is not case for `actions/cache` and `actions/cache/restore`, though. They share some semantics on their APIs (the same input controls opt-out from restoring in both of them), but who knows what wait us in the wild 😄

I'm happy to go for `RepositoryUses`, I don't have a strong opinion here. Let me know what would be your call 🙂

Reply:
Done [here](https://github.com/woodruffw/zizmor/pull/294/commits/85475ab269837f423dceab783fac81f838a20cee)